### DISCUSSION TITLE
Running multiple servers and/or generating openapi specifications with subsets of actions

### DISCUSSION THREAD

**Discussion by jeanibarz** [robocorp/robocorp#249] [Q&A]
Hey there,

I've got a bunch of actions—more than 30—that I've set up with OpenAI's CustomGPT. But it's not possible to use more than 30 at once. So, I'm thinking of splitting them into different groups, like one for pulling up context, another for managing emails, and a third for stuff like weather and maps. Each group of actions would be used on a single different GPT.

However, when I attempted to run these groups separately using different action servers, I encountered an error while attempting to run more than one server simultaneously. Seems like running multiple action servers at once might not be the way to go.

Any tips on how I can organize my actions better to make this work smoothly?

Thanks a bunch!

**Comment by fabioz** [comment]
@jeanibarz The latest release (0.0.28) added 2 things that may help here:

1. It's now possible to whitelist actions (so, you can just add the ones you're interested in):
https://github.com/robocorp/robocorp/blob/master/action_server/docs/guides/03-whitelist-on-start-or-import.md

2. It's now possible to deal with custom data, so, you could create a single action as the entry point and then it can delegate internal commands based on that (as you'd do on a command-line tool): https://github.com/robocorp/robocorp/blob/master/action_server/docs/guides/04-custom-data.md
